php_phongo.c

```diff
@@ -1963,44 +1975,69 @@ static mongoc_ssl_opt_t* php_phongo_make_ssl_opt(zval* zoptions TSRMLS_DC)
 
 	ssl_opt = ecalloc(1, sizeof(mongoc_ssl_opt_t));
 
+	/* If SSL options are set in the URL, we need to read them and set them on
+	 * the options struct so we can merge potential options from passed in
+	 * driverOptions (zoptions) */
+	if (mongoc_uri_get_ssl(uri)) {
+		php_phongo_mongoc_ssl_opts_from_uri(ssl_opt, uri);
+		any_ssl_option_set = true;
```

Comment:
My comments in https://github.com/mongodb/mongo-php-driver/pull/893#discussion_r204740634 still apply here. This may set `any_ssl_option_set` to true even if `php_phongo_mongoc_ssl_opts_from_uri()` doesn't actually do anything, because `mongoc_uri_get_ssl()` will return true when only `ssl=true` is set in the URI (with none of the other SSL-related options).

My suggestion was to either pass in the address of `any_ssl_option_set` to have `php_phongo_mongoc_ssl_opts_from_uri()` set it accordingly (as you do later in this very function for each option) or to just inline `php_phongo_mongoc_ssl_opts_from_uri()` here.

Reply:
I've done that now for ``pem_file || pem_pwd || ca_file`` as these are ``null`` if not set, but it's hard to say whether ``weak_cert_validation`` or ``allow_invalid_hostname`` are like that — as they're booleans defaulting to ``false``.